gst/nnstreamer/tensor_converter/tensor_converter.c

```diff
@@ -1121,36 +1121,49 @@ gst_tensor_converter_chain (GstPad * pad, GstObject * parent, GstBuffer * buf)
       }
       break;
     case _NNS_OCTET:
-      /** get frame size from the properties */
-      g_assert ((buf_size % frame_size) == 0);
-      frames_in = buf_size / frame_size;
+      if (gst_tensors_config_is_flexible (config)) {
+        /* update dimension with buffer size */
+        config->info.info[0].dimension[0] = buf_size;
```

Comment:
Is there no problem if the octet-stream has multiple memory chunks?

Reply:
I updated this PR to handle multi memories in in input buffer.